daemon/src/monitor.rs

```diff
@@ -899,14 +1060,15 @@ mod tests {
 
     impl Actor<stub::Client> {
         #[allow(clippy::type_complexity)]
-        fn for_test<const N: usize>(
+        async fn for_test<const N: usize>(
             event_channel: Box<dyn StrongMessageChannel<Event>>,
             subscriptions: [((Txid, Script), Vec<(ScriptStatus, Event)>); N],
         ) -> Self {
             Actor {
                 cfds: HashMap::default(),
                 event_channel,
                 client: stub::Client::default(),
+                db: SqlitePool::connect(":memory:").await.unwrap(),
```

Comment:
This shouldn't be necessary if you do it in the ctor because then you don't need to save the db as a field.